src/trace-server.ts

```diff
@@ -81,6 +89,9 @@ export class TraceServer {
             return;
         }
         if (pid) {
+            // Remove all existing listeners to avoid unnecessary pop-ups
+            this.server?.removeAllListeners();
```

Comment:
Another use of `this.server.removeAllListeners()` - I am less worried about this one since the goal is to kill the server. But if there's an easy way to remove the listener(s) we ourselves registered, it would still be preferable. 

Alternatively, we might leave the "exit event listener" in place and let it hit when the server is killed, thus de-registering itself (applied using `.once()`). If it's the only listener of ours that's expected to be present, I think this works.

Reply:
I'll use a specific handler.